mlir/include/mlir/Dialect/XeGPU/uArch/uArch.h

```diff
@@ -0,0 +1,389 @@
+// - the unit of computation,
+// - the description of the instruction
+// The information is represented as strings
+// For example, the information about an instruction can be represented as:
+// Instruction info = {"dpas", "0x83", "matrix", "simd", "subgroup", "tile",
+// "Dot Product Accumulate Systolic  (DPAS) is a matrix multiply-add
+// operation"};
+
+// The primary purpose of Instruction struct is to provide a generic way to
+// represent information about an instruction and to use this information to
+// generate the uArch. Specifc instruction in a uArch can inherit from this
+// struct and add more fields as needed
+
+struct Instruction {
+  std::string name;
+  std::string description;
+  std::string opcode;
+  FunctionalUnit functional_unit;
+  InstructionType type;
+  InstructionScope scope;
+  UnitOfComputation unit_of_computation;
+
+  // @TODO: Add more fields as needed
+  // std::string latency;
+  // std::string throughput;
+  // std::string pipeline;
+  // std::string resource;
+  // std::string comment;
+  Instruction(std::string name, std::string desc, std::string opcode,
+              FunctionalUnit fu, InstructionType itype, InstructionScope sc,
+              UnitOfComputation uoc)
+      : name(std::move(name)), description(std::move(desc)),
+        opcode(std::move(opcode)), functional_unit(fu), type(itype), scope(sc),
+        unit_of_computation(uoc) {}
+
+  virtual ~Instruction() = default;
+};
+
+// A struct to represent register file information
+struct RegisterFileInfo {
+  uint32_t size;                 // size per register in bits
+  std::vector<std::string> mode; // e.g., "small", "large" GRF modes
+  std::vector<uint32_t>
+      num_regs_per_thread_per_mode; // number of registers per thread per mode
+  uint32_t num_banks;
+  uint32_t bank_size;
+
+  // Constructor
+  RegisterFileInfo() = default;
+  RegisterFileInfo(uint32_t size, const std::vector<std::string> &mode,
+                   const std::vector<uint32_t> &numRegs, uint32_t num_banks,
+                   uint32_t bank_size)
+      : size(size), mode(mode), num_regs_per_thread_per_mode(numRegs),
+        num_banks(num_banks), bank_size(bank_size) {}
+};
+
+// A struct to represent cache information
+struct CacheInfo {
+  uint32_t size;
+  uint32_t line_size;
+  // At which component level the cache is shared
+  uArchHierarchyComponent component;
+  // uint32_t associativity;
+  // uint32_t num_banks;
+  // uint32_t bank_size;
+  // uint32_t num_ports;
+  // uint32_t port_width;
+  // uint32_t bank_conflicts;
+  // Constructor
+  CacheInfo(uint32_t size, uint32_t line_size,
+            const uArchHierarchyComponent &component)
+      : size(size), line_size(line_size), component(component) {}
+};
+
+// A struct to represent the uArch
+// This struct is used to represent the microarchitecture of a target device
+// The uArch includes:
+// - the name of the uArch,
+// - the description of the uArch,
+// - the range of tiles supported by the uArch,
```

Comment:
I don't think there is such a uArch level block size support. These are always per instructions. 2dload, 2dstore, and DPAS operations support different block size.